
m bc510378b38244fa4535dbdbc23b92621dc24663 Mon Sep 17 00:00:00 2001
From: "Owen W. Taylor" <otaylor@fishsoup.net>
Date: Fri, 27 Jun 2014 14:33:01 -0400
Subject: Switch to using ClutterStage::after-paint

The experimental API clutter_stage_set_paint_callback() is replaced
with an ::after-paint signal as of Clutter 1.19.5.

Index: mutter-3.12.2/configure.ac
===================================================================
--- mutter-3.12.2.orig/configure.ac
+++ mutter-3.12.2/configure.ac
@@ -75,7 +75,7 @@ MUTTER_PC_MODULES="
    cairo >= 1.10.0
    gsettings-desktop-schemas >= 3.7.3
    xcomposite >= 0.2 xfixes xrender xdamage xi >= 1.6.0
-   $CLUTTER_PACKAGE >= 1.15.90
+   $CLUTTER_PACKAGE >= 1.19.5
    cogl-1.0 >= 1.17.1
    upower-glib >= 0.9.11
    gnome-desktop-3.0
Index: mutter-3.12.2/src/compositor/compositor.c
===================================================================
--- mutter-3.12.2.orig/src/compositor/compositor.c
+++ mutter-3.12.2/src/compositor/compositor.c
@@ -566,10 +566,8 @@ meta_compositor_manage_screen (MetaCompo
 
   info->stage = clutter_stage_new ();
 
-  clutter_stage_set_paint_callback (CLUTTER_STAGE (info->stage),
-                                    after_stage_paint,
-                                    info,
-                                    NULL);
+  g_signal_connect (CLUTTER_STAGE (info->stage), "after-paint",
+                    G_CALLBACK (after_stage_paint), compositor);
 
   clutter_stage_set_sync_delay (CLUTTER_STAGE (info->stage), META_SYNC_DELAY);
 
